features/enlinkd/tests/src/test/java/org/opennms/netmgt/enlinkd/Nms17216EnIT.java

```diff
@@ -745,10 +748,142 @@ public void testNetwork17216CdpTopology() throws Exception {
                 
+        assertTrue(m_cdpTopologyService.hasUpdates());
+        System.err.println("--------Printing new start----------");
+        m_linkd.runTopologyUpdater(ProtocolSupported.CDP);
+        System.err.println("--------Printing new end----------");
+        assertTrue(!m_cdpTopologyService.hasUpdates());
+        assertTrue(!m_cdpTopologyService.parseUpdates());
+        System.err.println("--------no updates start----------");
+        m_linkd.runTopologyUpdater(ProtocolSupported.CDP);
+        System.err.println("--------no updates end----------");
+        m_cdpTopologyService.updatesAvailable();
+        assertTrue(m_cdpTopologyService.parseUpdates());
+        assertTrue(!m_cdpTopologyService.hasUpdates());
     }
+    
+    @Test
+    @JUnitSnmpAgents(value={
+            @JUnitSnmpAgent(host=SWITCH1_IP, port=161, resource=SWITCH1_SNMP_RESOURCE),
+            @JUnitSnmpAgent(host=SWITCH2_IP, port=161, resource=SWITCH2_SNMP_RESOURCE),
+            @JUnitSnmpAgent(host=SWITCH3_IP, port=161, resource=SWITCH3_SNMP_RESOURCE)
+    })
+    public void testNetwork17216lldpTopology() throws Exception {
+        m_nodeDao.save(builder.getSwitch1());
+        m_nodeDao.save(builder.getSwitch2());
+        m_nodeDao.save(builder.getSwitch3());
+        
+        m_nodeDao.flush();
+
+        m_linkdConfig.getConfiguration().setUseBridgeDiscovery(false);
+        m_linkdConfig.getConfiguration().setUseCdpDiscovery(false);
+        m_linkdConfig.getConfiguration().setUseOspfDiscovery(false);
+        m_linkdConfig.getConfiguration().setUseLldpDiscovery(true);
+        m_linkdConfig.getConfiguration().setUseIsisDiscovery(false);
+
+        assertTrue(m_linkdConfig.useLldpDiscovery());
+        assertTrue(!m_linkdConfig.useCdpDiscovery());
+        assertTrue(!m_linkdConfig.useOspfDiscovery());
+        assertTrue(!m_linkdConfig.useBridgeDiscovery());
+        assertTrue(!m_linkdConfig.useIsisDiscovery());
+
+        final OnmsNode switch1 = m_nodeDao.findByForeignId("linkd", SWITCH1_NAME);
+        final OnmsNode switch2 = m_nodeDao.findByForeignId("linkd", SWITCH2_NAME);
+        final OnmsNode switch3 = m_nodeDao.findByForeignId("linkd", SWITCH3_NAME);
+
+        assertTrue(m_linkd.scheduleNodeCollection(switch1.getId()));
+        assertTrue(m_linkd.scheduleNodeCollection(switch2.getId()));
+        assertTrue(m_linkd.scheduleNodeCollection(switch3.getId()));
+
+        assertTrue(m_linkd.runSingleSnmpCollection(switch1.getId()));
+        assertEquals(4, m_lldpLinkDao.countAll());
+        assertTrue(m_lldpTopologyService.hasUpdates());
+        
+        assertTrue(m_linkd.runSingleSnmpCollection(switch2.getId()));
+        assertEquals(10, m_lldpLinkDao.countAll());
+        assertTrue(m_lldpTopologyService.hasUpdates());
+        
+        // Now we support by default configuration: BRIDGE, CDP, ISIS, LLDP and OSPF
+        assertEquals(5, m_topologyDao.getSupportedProtocols().size());
```

Comment:
This is out of place - why are we asserting that the set of supported protocols in the topologyDao has these 5 protocols in a test called `testNetwork17216CdpTopology` in the `Nms17216EnIT.java`?

Reply:
I have updated code su support reload event now.
Added tests..too.

commit 4e6b0023e2e14278cee91dc74c37abdeea40fa29 (HEAD -> jira/HZN-1393)
Author: Antonio <rssntn67@yahoo.it>
Date:   Wed Jan 16 19:34:31 2019 +0100

    Added reload configuration for enlinkd

    Supported reload event for enlinkd.
    Added reloadConfig to update configuration and reload daemon
    Also let spring instantiate Updaters and BridgeDiscovery
    Renamed common classes to NodeCollector and Topologyupdater